crates/ruff_linter/src/rules/flake8_simplify/rules/open_file_with_context_handler.rs

```diff
@@ -114,6 +114,30 @@ fn match_exit_stack(semantic: &SemanticModel) -> bool {
     false
 }
 
+/// Return `true` if the current expression is nested in a call to one of the
+/// unittest context manager methods: `cls.enterClassContext()`,
+/// `self.enterContext()`, or `self.enterAsyncContext()`.
+fn match_unittest_context_methods(semantic: &SemanticModel) -> bool {
+    let Some(expr) = semantic.current_expression_parent() else {
+        return false;
+    };
+    let Expr::Call(ast::ExprCall { func, .. }) = expr else {
+        return false;
+    };
+    let Expr::Attribute(ast::ExprAttribute { attr, value, .. }) = func.as_ref() else {
+        return false;
+    };
+    if attr != "enterClassContext" && attr != "enterContext" && attr != "enterAsyncContext" {
+        return false;
+    }
+    let Expr::Name(ast::ExprName { id, .. }) = value.as_ref() else {
+        return false;
+    };
+    id == "cls" && attr == "enterClassContext"
+        || id == "self" && attr == "enterContext"
+        || id == "self" && attr == "enterAsyncContext"
```

Comment:
```suggestion
    matches!(
        (id, attr),
        ("cls", "enterClassContext") | ("self", "enterContext" | "enterAsyncContext")
    )
```

Reply:
Thanks, that’s much better. Still a Rust noob here.